Source/SwiftLintFramework/Extensions/String+SwiftLint.swift

```diff
@@ -23,11 +23,11 @@ extension String {
     }
 
     internal func isUppercase() -> Bool {
-        return self == uppercased()
+        return CharacterSet.uppercaseLetters.isSuperset(ofCharactersIn: self)
```

Comment:
I agree that this might be an issue, but currently we only use this for single characters.

Reply:
> currently we only use this for single characters.

If you mean single scalars, it can be ignored, but if you mean single grapheme clusters (what Swift calls `Character`), this will fail.

é (e + ◌́) is one cluster, but two scalars.
```swift
"é".isUppercase() // → true?!?
```
This happens because ◌́ is not a member of `.lowercaseLetters`.

***

The old fails too for Unicode [General Category](https://en.wikipedia.org/wiki/Unicode_character_property#General_Category) Lt, such as “ǅ” (as one scalar)
```swift
"ǅ".isUppercase() // → false?!?
```
This happens because `"ǅ".uppercased()` yields `DŽ`
(Technically Dž is titlecase, not uppercase, but this should be looking for either.)

***

This would be safer (still assuming only one character):
```swift
return unicodeScalars.contains(where: CharacterSet.uppercaseLetters.contains) // Lu or Lt
```
If this is only intended for a single character, maybe extending `Character` or `UnicodeScalar` is clearer than extending `String`?

Reply:
> If this is only intended for a single character, maybe extending Character or UnicodeScalar is clearer than extending String?

I agree.